cohorts/load.py

```diff
@@ -397,14 +412,17 @@ def is_lambda(func):
             for key, value in on.iteritems():
                 col, df = apply_func(on=value, col=key, df=df)
                 cols.append(col)
-            return (cols, df)
         if type(on) == list:
             cols = []
             for i, elem in enumerate(on):
                 col = elem.__name__ if not is_lambda(elem) else "column_%d" % i
                 col, df = apply_func(on=elem, col=col, df=df)
                 cols.append(col)
-            return (cols, df)
+
+        if (rename_cols):
+            df = df.rename(columns=_strip_column_names(df.columns,
```

Comment:
I apologize for the confusing nature of this code, but I think this will break in some instances when the name of the function, e.g. `missense_snv_counT` gets stripped here, however it doesn't get stripped in the `cols` variable.

Maybe, instead of putting this `if (rename_cols)` here, you should put it inside `apply_func` above, replacing `df[col] = df.apply(func, axis=1)`?

Something to replace both the column name and the dataframe e.g.:

```
applied = df.apply(func, axis=1)
if (rename_cols):
    df.rename(...)
    col = _strip_column_name(...)
df[col] = applied
```